More rough and shady jobs by @Pointedstick

MCOfficer opened the pull request:
Ref: endless-sky/endless-sky#3235

> Added some shady passenger transport jobs and a few flavorful jobs available in dangerous parts of the galaxy, including an opportunity to gamble with pirates.

tehhowch committed:
Check if the picked boarding/assisting mission has space before offering

 - Use Mission::HasSpace(ship) rather than check in MainPanel::StepEvents
 - Update blocked mission handling to use the proper mission list
 - Use correct cargo / bunk numbers for "Blocked" messages if boarding
 - Set proper autoconditions when boarding

tehhowch committed:
Add mission cargo from boarding/assisting missions only to the flagship

 - Missions that add or remove outfits use the proper CargoHold.

tehhowch committed:
Extend LocationFilter's supported criteria

  Missions that trigger on boarding or assisting can now reference the ship's
    - category
      - 'and'ing a category is not possible (since multi-category ships are not possible).
      - e.g. `category "Heavy Warship" "Medium Warship"` matches any ship that is either a Heavy Warship or a Medium Warship
    - outfits
      - 'and'ing outfits is supported
      - e.g. `outfits "Electron Beam" "Electron Turret"` matches any ship that has at least an electron beam or an electron turret installed (or in cargo)
    - attributes
      - 'and'ing attributes is supported
      - e.g. `attributes "cloaking fuel"` matches any ship that has a positive "cloaking fuel" attribute, after all its installed outfits' effects are applied

  - Outfit and attribute references return true if the ship has a positive value for the provided token - a ship with a negative or zero-valued attribute/outfit count will not match that attribute or outfit.

  - Outfits can also be used for planets to restrict by the items available in its outfitter.
    e.g. `outfits "Stack Core"` would match planets Solace, Wayfarer, and Zug, after the Stack Core is available for purchase

tehhowch committed:
Use a preference to control fighter repair strategy

 - The player can choose between least need (series) and most need (parallel). Maintains the default strategy of least need for AI ships and players who have not set the preference.

Pointedstick committed:
More rough and shady jobs

Pointedstick committed:
The player is probably not a drywaller

Pointedstick committed:
- "his his" -> "his"
- Also tone down the violence a bit in how the combat is described, to
better match the tone of the rest of the game

Pointedstick committed:
- Addressed review comments
- Updated to new illegal syntax introduced with
https://github.com/endless-sky/endless-sky/commit/03b24218338c5377d6a0849a58e2b3a52498b75f
- Gave illegal job tags to the existing drug running missions, for
future flavor possibilities

Pointedstick committed:
Quoting a couple more tokens

Pointedstick committed:
Pirates probably wouldn't favor fancy energy weapons

Pointedstick committed:
"south mining" -> "south" "mining"

Pointedstick committed:
Grammar fix

tehhowch committed:
Place NPCs from boarding missions while in-flight

 - Expose a pointer to the recently-accepted boarding mission
 - Move "Place" code for NPCs to public method
 - Check for an active boarding mission in SpawnFleets

Pointedstick committed:
Review feedback, omissions, tweaks

Pointedstick committed:
Make the Pirate Gambling mission available to more early-game players
and for longer

tehhowch committed:
Update "require <outfit> n" handling of out-of-system escorts

 - Addresses the issues raised in endless-sky#3034, endless-sky#3036 and endless-sky#3075 w.r.t. outfits to be transferred

pscamman committed:
Adds support for weapons that use heat as a resource rather than a cost.

 - Simply prevents a weapon from firing if it would reduce your heat below 0
 - To make use of this, create weapons with negative heat costs

Pointedstick committed:
Fix typo and improve wording a tad bit

Amazinite commented:
I think this PR can wait until we get some sort of random payment method (endless-sky#3743)

Amazinite committed:
Normalized spelling of gases

Nescio0 committed:
update 'hull disabled at' tooltips

Lorantine committed:
typo

elgeonmb committed:
Adds extra response to the "Alondo is gay" revelation (#3705)

* lets you come out to freya winters

* changes freya's wording to be more freya

* Changed the dialog choice to flow better with the conversation

czartrak committed:
Center Pointedstick's Vanguard

Ref: #3954

* Delete pointedstick_vanguard.png

* Add files via upload

* Update persons.txt

warp-core committed:
Typo fix

tehhowch committed:
Add configuration files for CI tools

https://github.com/endless-sky/endless-sky/pull/2786

* Add configuration files for CI tools
 - .travis.yml for build verification of linux executables
 - .appveyor.yml for build verification of Windows executables
    + .winmake for "headless" builds of Endless Sky using only MinGW

* Update .gitignore to exclude more local-only files
 - Adding the CI files so local modifications can be made without committing them back to upstream

* Distribute an .editorconfig file to help maintain repo style
 - Some IDEs / editors have native support, others (Notepad++, Atom, VS Code, Codeblocks) have plugins available

Pointedstick committed:
Remove redundant line

Pointedstick committed:
We still have shotguns in the 31st century, and they're still
devastating

Pointedstick committed:
Correct grammar

Ferociousfeind committed:
Add hostile militia retorts

#3710

czartrak committed:
Assassination pirate jobs

#3984

TheUnfetteredOne committed:
New spaceport missions

#3151

Amazinite committed:
Blockade running missions

#3298

Amazinite committed:
Hai spaceport missions

#3781

Amazinite committed:
Terraforming Tundra

#3216

Amazinite committed:
Typo fix (@Proont)

Pointedstick committed:
Add a high-tech image for Lagrange

#3897

Fzzr committed:
Add a pirate mission to capture a rogue smuggler

#3475

Vilhelm16 committed:
Added unique explosions for Pug ships

#3957

Brick63 committed:
Added a Remnant H2H weapon

#3960

Brick63 committed:
Added three new Hai ships

#3314

Pointedstick committed:
Tweak default ship and variant loadouts to reflect missile changes

#3922

Amazinite committed:
Error fixes

Pointedstick committed:
Altered the shield/hull balance of the Corvette and Firebird

#3411

tiennou committed:
Fix XCode libraries

https://github.com/endless-sky/endless-sky/pull/3808

* Fix path to brew's SDL2.dylib install 

* Remove unused framework path

* Don't add some Xcode directory to the header search path

"XCode/mumble/jumble/usr/include" should be equivalent to /usr/include
(and the build system takes care of prepending the SDKROOT).

Having this first makes the OS-provided non-jpeg-turbo library to be
picked up.

tehhowch committed:
Enable save-aware data parsing without modifying the save

https://github.com/endless-sky/endless-sky/pull/3994

* Allow parsing game data for content errors

 - Specify the `-p` or `--parse-save` command-line option to load the most recent save and allow the game to perform reference-checking.
 This option exits before initializing SDL (e.g. headless operation), and does not save the file before exiting.
 - Updated the man page and help text as well

* Include whitespace warnings in DataFile::Load

 - Print warnings to stderr and errors.txt when spaces and tabs are mixed
 - Provides a warning if a line starts with a space, and then tabs are later used in the file for indenting
 - Provides a line-specific warning if spaces and tabs are intermixed on the same line.
 - Fix existing mixed whitespace indentation

* Extend CI tooling with data parsing test

 - add bash script for Travis CI
 - add PS script for AppVeyor CI

The scripts invoke the new `-p` command-line flag and test whether the `errors.txt` file was created.
If so, this is taken to mean that a parsing error occurred. The scripts can be executed locally by passing the appropriate executable as the argument to the script.

* Add OSX build environment

 - Homebrew installations of the developer dependencies (per the developer readme).
   (This will eventually result in a broken build when a core dependency publishes a new point release, at which point the XCode project will need to be updated.)

* Avoid automatically saving upon game load 

 - Closes https://github.com/endless-sky/endless-sky/issues/3510
 - Only allow saving after the player has explicitly "entered" their savegame
   Players explicitly 'enter' their savegame via the keypress / button:
   - 'Enter Ship' (after first launching the game)
   - 'Load Game' (after pressing Load/Save and then selecting a different save)

In other words, if you tell the game, "Yes, this is the save I want to play," its save file will be updated when you close the game.
If you didn't explicitly tell the game that, then the save file will not be modified.

tehhowch committed:
Remove compiler warnings issued by clang-902.0.39.2

 - Prefer auto for small-scope vars with long type names

Amazinite committed:
Capitalized "Wasp" in its description (@Nescio0)

tehhowch committed:
Ships cannot have negative outfit counts

This is a core assumption of ShopPanel code, attribute combination, and so on.

Nescio0 committed:
Ships without fuel to jump display a "no fuel?" warning in Shipyard

https://github.com/endless-sky/endless-sky/pull/3843
* Add `no fuel?` warning

* Add `no fuel?` tooltip

* check against jump fuel

* updated tooltip

* `!canBeCarried` order

* combined lines

* improved no fuel tooltip

* changed to "insufficient fuel"

* JumpFuel()

* fuelCapacity

Amazinite committed:
Ignore some additional files and folders

tehhowch committed:
Support launch effects for carried ships when departing from their carrier (#3602)

* Support "launch effect" child node for carrier bays

 - Allows specifying multiple effects, and a count of how many of each, e.g.
    "launch effect" "door sparks" 5
    "launch effect" "decompression"

* Provide a fallback effect for certain bay types

For non-automaton ships, any internal (i.e. pressurized) bays will use a default effect if none were specified

* Remove unnecessary includes

 - Angle & Point includes were needed before `Visual` took the responsibility for placement of effects

tehhowch committed:
Issue deprecation warnings & simplify some code (#4000)

* Fix osx compile warning

* Emit warnings for deprecated turret specs and homing w/o tracking

* Require ranges / radii to be non-negative

* Replace implicit defaults with explicit values

* Handle deprecated "cargo scan" and "outfit scan"

* Replace locally repeated code with local lambda

Amazinite committed:
Added wartime jobs for the Free Worlds

https://github.com/endless-sky/endless-sky/pull/3330
* Added wartime jobs for the Free Worlds

* Feedback changes

* Implemented new waypoints text replacement

* Using the new neighbor location filter

* Typo fixes

tehhowch committed:
More syntax warnings

 - If a mission source, destination, waypoint, or stopover directly specify an object, only the first specified is used.
   Provides a warning to the content creator that additional tokens are ignored, and so is any location filter, if a specific token is used.
 - Warns about using 'illegal' and 'stealth' as children of `cargo`

tehhowch committed:
Update data files to reflect proper syntax

tehhowch committed:
Handle "outfit <outfit> 0" while loading

tehhowch committed:
Remove unneeded check

Nescio0 committed:
Organized ship, outfit, and effect attributes to be consistent

#3868

Brick63 committed:
Added sprite for the Quarg outfits

#3536

Nescio0 committed:
Alphabetize variant definitions

https://github.com/endless-sky/endless-sky/pull/4008

tehhowch committed:
Add issue templates

https://github.com/endless-sky/endless-sky/pull/3986

* Add issue templates

* Incorporate feedback

* Add link to Issues page

Brick63 committed:
Add "leaks" to Hai ships

https://github.com/endless-sky/endless-sky/pull/3998

 - Uses the existing game's leak effects

Nescio0 committed:
Adjust Kestrel customization outcomes

https://github.com/endless-sky/endless-sky/pull/3976

Base model:
 - Adds 2 Tactical scanners
 - Increases base shield and hull
 - Reduces required crew
 - Reduces drag and increases heat dissipation
 - minor bump to stock weapon capacity

"More Engines"
 - Reduces drag further
 - More fuel capacity
 - +30 engine capacity (instead of +20)

"More Shields"
 - +3000 shields (instead of +2700)
 - +1500 hull (instead of +1300)

"More Weapons"
 - +40 weapon capacity (instead of +20)
 - +12 gun ports (18 total instead of 6)

tehhowch committed:
Add a bit more quick info to the readme

tehhowch committed:
Better highlight the low memory / space requirements

Pointedstick committed:
- Address review comments by simplifying the writing
- Add a new shady passenger double-cross mission that's now possible

Pointedstick committed:
Merge branch 'master' into more-rough-jobs

Pointedstick committed:
Address review comments

Pointedstick committed:
Luxury Accommodations and high-class passenger jobs (#3359)

* Luxury Accomodations and high-class passenger jobs

* Revert change to require Luxury Accommodations for Deep scientists;
they're probably as strapped for cash 1000 years in the future as they
are today

* Match existing style for dashes used instead of commas

* Update image to have a nice wooden bed and a patterned bedspread

* - Gave Luxury Accommodations to the Blackbird by default, to match its description
- Added luxury variants for the Arrow, Bounder, and Flivver
- Added a few fleet options for luxury ships in the Core and the North

* Reduced the number of passenger jobs that are gated behind owning Luxury
Accomodations

* Restored original formatting for Hai tourist jobs' payment syntax

* Typo

* Adjust Quantum Keystone description to hint at the jobs it helps unlock

* Bring some of the variants up-to-date with recent missile changes

Amazinite committed:
Added more regional flavor jobs

#3220

Amazinite committed:
Rename syndicate world jobs.txt to syndicate jobs.txt

to fit the naming system of the other regional jobs

tehhowch committed:
Print warnings for ships with negative space

 - Scans for the "big 4": outfit, weapon, engine, and cargo
 - If one or more is negative, prints the outfits & attributes to stderr, along with the model name (and ship name, if defined)
 - Most useful for checking stock variants for issues following changes to outfit attributes.

Pointedstick committed:
Correct new luxury ship variant definitions

Pointedstick committed:
Fix some variants that had negative outfit space

Amazinite committed:
Handled negative outfit space on the Protector (laser) differently

Reducing shielding instead of cooling so that the ship can still reasonably function

tehhowch committed:
Address some negative outfit spaces in person ships

Nescio0 committed:
Re-order Carrier's HAM turret (#3992)

Pointedstick committed:
Address review comments

Pointedstick committed:
Merge branch 'master' into more-rough-jobs

Pointedstick committed:
Tweak the double-cross mission some more

Pointedstick committed:
Correctly end the double cross mission if you bribe the mercs
